airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -0,0 +1,148 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import tempfile
+
+import yaml
+from kubernetes import client, config
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+
+
+def _load_body_to_dict(body):
+    try:
+        body_dict = yaml.safe_load(body)
+    except yaml.YAMLError as e:
+        raise AirflowException("Exception when loading resource definition: %s\n" % e)
+    return body_dict
+
+
+class KubernetesHook(BaseHook):
+    """
+    Creates Kubernetes API connection.
+
+    :param conn_id: the connection to Kubernetes cluster
+    """
+
+    def __init__(
+        self,
+        conn_id="kubernetes_default"
+    ):
+        self.connection = self.get_connection(conn_id)
+        self.extras = self.connection.extra_dejson
```

Comment:
We should not have "connection to DB" when instantiating Hook. The main reason is DAG files are parsed at regular interval and this can cause an increase in DB connections

Reply:
Similar to https://github.com/apache/airflow/blob/master/airflow/providers/microsoft/azure/hooks/azure_cosmos.py

Reply:
fixed